File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/functional/StreamUtilTest.java
package ch.hsr.ifs.iltis.core.core.tests.functional;

import java.util.HashMap;
import java.util.Map;

import org.junit.Assert;
import org.junit.Test;

import ch.hsr.ifs.iltis.core.core.functional.Functional;
import ch.hsr.ifs.iltis.core.core.functional.StreamUtil;


public class StreamUtilTest {

   @Test
   public void toMapCollector() {
      String[] as = { "foo", "bar", "baz" };
      Integer[] bs = { 1, 2, 3, 4 };

      Map<String, Integer> actual = Functional.zip(as, bs).collect(StreamUtil.toMap());

      Map<String, Integer> expected = new HashMap<>();
      expected.put(as[0], bs[0]);
      expected.put(as[1], bs[1]);
      expected.put(as[2], bs[2]);
      expected.put(null, bs[3]);

      Assert.assertEquals(expected, actual);
   }

}

File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/functional/StreamUtilTest.kt
package ch.hsr.ifs.iltis.core.core.tests.functional

import org.junit.Test
import ch.hsr.ifs.iltis.core.core.functional.Functional
import ch.hsr.ifs.iltis.core.core.functional.StreamUtil
import org.junit.Assert.assertEquals


class StreamUtilTest {

	@Test
	fun `toMap() Collector sets missing key to 'null'`() {
		val keys = arrayOf("foo", "bar", "baz")
		val vals = arrayOf(1, 2, 3, 4)

		val actual = Functional.zip(keys, vals).collect(StreamUtil.toMap())

		val expected = mapOf(keys[0] to vals[0], keys[1] to vals[1], keys[2] to vals[2], null to vals[3])

		assertEquals(expected, actual);
	}

}
